src/doc/grammar.md

```diff
@@ -502,6 +502,12 @@ tuple_expr : '(' [ expr [ ',' expr ] * | expr ',' ] ? ')' ;
 unit_expr : "()" ;
 ```
 
+### Never expressions
+
+```antlr
+never_expr : "!" ;
```

Comment:
This still isn't right. `!` isn't an expression, just a type (although the sections for primitive types is nothing but FIXMEs at the moment).